ciao-controller/internal/datastore/datastore.go

```diff
@@ -245,6 +250,34 @@ func (ds *Datastore) initImages() error {
 	return nil
 }
 
+func (ds *Datastore) initWorkloads() error {
+	ds.workloadsLock = &sync.RWMutex{}
+	ds.workloads = make(map[string]types.Workload)
+	workloads, err := ds.db.getWorkloads()
+	if err != nil {
+		return errors.Wrap(err, "error getting workloads from database")
+	}
+
+	for _, wl := range workloads {
+		ds.workloads[wl.ID] = wl
+
+		if wl.Visibility == types.Public {
+			ds.publicWorkloads = append(ds.publicWorkloads, wl.ID)
+		}
```

Comment:
Can workloads be public and belong to a tenant?

Reply:
Inconsistently. I've fixed it so that it they're exclusive.